kvbc/src/categorization/blocks.cpp

```diff
@@ -41,31 +43,21 @@ BlockMerkleInput RawBlock::getUpdates(const std::string& category_id,
                                       const BlockMerkleOutput& update_info,
                                       const BlockId& block_id,
                                       const std::shared_ptr<storage::rocksdb::NativeClient>& native_client) {
-  // For old serialization
-  using namespace concord::kvbc::v2MerkleTree;
   BlockMerkleInput data;
-  // Iterate over the keys:
-  // if deleted, add to the deleted set.
-  // else generate a db key, serialize it and
-  // get the value from the corresponding column family
+  detail::BlockMerkleCategory cat{native_client};
   for (auto& [key, flag] : update_info.keys) {
     if (flag.deleted) {
       data.deletes.push_back(key);
       continue;
     }
-    // E.L see how we can optimize the sliver temporary allocation
-    auto db_key =
-        v2MerkleTree::detail::DBKeyManipulator::genDataDbKey(Key(std::string(key)), update_info.state_root_version);
-    auto val = native_client->get(category_id, db_key);
+    // get value of the key for a version from storage via the category
+    auto val = cat.get(key, block_id);
     if (!val.has_value()) {
-      throw std::runtime_error("Couldn't find value for key");
+      auto msg = std::string("Couldn't find value for key [") + key + std::string("] (Merkle category)");
```

Comment:
I think this is an assert situation.

Reply:
Good catch. I agree. A key must exist if its in an existing block.